Content.Server/PDA/Ringer/RingerSystem.cs

```diff
@@ -0,0 +1,182 @@
+    {
+
+        private readonly Dictionary<string, string> _pathIndex = new Dictionary<string, string>();
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            //General Event Subscriptions
+            SubscribeLocalEvent<RingerComponent, ComponentInit>(RandomizeRingtone);
+            //RingerBoundUserInterface Subscriptions
+            SubscribeLocalEvent<RingerComponent, RingerSetRingtoneMessage>(RingerUpdateRingtoneDriver);
+            SubscribeLocalEvent<RingerComponent, RingerPlayRingtoneMessage>(RingerPlayRingtone);
+            SubscribeLocalEvent<RingerComponent, RingerRequestUpdateInterfaceMessage>(UpdateRingerUserInterfaceDriver);
+
+            //Dictionary Path Addition
+            for(int indexCount = 0; indexCount < 12; indexCount++)
+            {
+                _pathIndex.Add(((Note)indexCount).ToString(), string.Concat("/Audio/Effects/RingtoneNotes/", ((Note)indexCount).ToString().Replace("sharp", "#").ToLower()) + ".ogg");
+            }
+        }
+
+        //Event Functions
+
+        private void RingerPlayRingtone (EntityUid uid, RingerComponent ringer, RingerPlayRingtoneMessage args)
+        {
+            ringer.IsPlaying = true; //Allows the overridden Update function to play the ringtone based off of the ringer component's stored ringtonePathArr
+            UpdateRingerUserInterface(ringer);
+        }
+
+        private void UpdateRingerUserInterfaceDriver(EntityUid uid, RingerComponent ringer, RingerRequestUpdateInterfaceMessage args)
+        { //Seprate Drivers are needed because UpdateRingerUserInterface() and UpdateRingerComponent() are called by other functions aside from the events themselves
+            UpdateRingerUserInterface(ringer);
+        }
+
+        private void RingerUpdateRingtoneDriver(EntityUid uid, RingerComponent ringer, RingerSetRingtoneMessage args)
+        {
+            Note[] _RingtoneNoteArgumentArray = new Note[4];
+
+            for(int noteCount = 0; noteCount < 4; noteCount++) // Converts incoming string arguments into Note form and checks for nulls
+            {
+                try
+                {
+                string _currentNoteString = args.RingtoneStringArgumentArray[noteCount].Replace("#", "sharp");
+                _RingtoneNoteArgumentArray[noteCount] = (Note)Enum.Parse(typeof(Note), _currentNoteString);
+                }
+                catch
+                {
+                    return;
+                }
+            }
+
+            UpdateRingerComponent(ringer, _RingtoneNoteArgumentArray);
+        }
+
+        public void RandomizeRingtone(EntityUid uid, RingerComponent ringer, ComponentInit args) //Randomizes the ringtone
+        {
+            Note[] _randomizedRingerNoteArr = new Note[4];
+            var randomNote = new Random();
+
+            for(int count = 0; count < 4; count++)
+            {
+                int randomNoteIndex = randomNote.Next(Enum.GetNames(typeof(Note)).Length - 1);
```

Comment:
Just hardcode the length to 12 instead of having to parse the enum every single time for every ringer.